astropy/io/fits/hdu/base.py

```diff
@@ -888,7 +888,7 @@ def match_header(cls, header):
         # The check that 'GROUPS' is missing is a bit redundant, since the
         # match_header for GroupsHDU will always be called before this one.
         if card.keyword == 'SIMPLE':
-            if 'GROUPS' not in header and card.value == False:
+            if 'GROUPS' not in header and card.value is False:
```

Comment:
@mhvk - I kept this as is, as raising an ``InvalidHDUException`` seems to be better for empty values.